socorro/schemas/validate_processed_crash.py

```diff
@@ -95,22 +100,42 @@ def match_key(schema_key, document_key):
     :returns: boolean
 
     """
+    # This is the easy case where the two keys match
     if "(re:" not in schema_key:
         return schema_key == document_key
 
-    schema_parts = split_path(schema_key)
-    doc_parts = split_path(document_key)
-
-    for schema_part, doc_part in zip_longest(schema_parts, doc_parts, fillvalue=""):
-        if "(re:" in schema_part:
-            pattern_re = compile_pattern_re(schema_part[4:-1])
-            if not pattern_re.match(doc_part):
-                return False
-
-        elif schema_part != doc_part:
-            return False
-
-    return True
+    # This case is hard because one of the parts of the key can be a regex. So what we
+    # do is convert the entire schema key into a regex by doing some sleight-of-hand
+    # with the pattern_properties regexes particularly with the ^ and $. Then we use
+    # that regex to match the document_key. That handles schema keys like:
+    #
+    # .telemetry_environment.settings.userPrefs.(re:^.+$)
+    #
+    # which needs to match something like this:
+    #
+    # .telemetry_environment.settings.userPrefs.browser.startup.homepage
+    #
+    # because the keys have dots, too.
+    #
+    # Another way to do this matching thing is to store the path as a list of string
+    # parts rather than a .-delimited string, but that involves rewriting a bunch of
+    # stuff about paths and traversals and such.
+    schema_parts = []
```

Comment:
This is kind of ugly, but as I say in the comment, the right fix is to redo paths everywhere and I didn't want to tackle that in this PR. I'll write up a follow-up bug to do that.